### PR TITLE
Add tests for grdfilter

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request adds tests for the `grdfilter` function in `grdops.py`.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

**Comment by seisman** [comment]
@willschlitzer Just a quick comment. When opening a PR, please use the PR template, so we can have the PR checklist.

**Comment by willschlitzer** [comment]
@seisman Sorry about that.

File: pygmt/tests/test_grdfilter.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(registration="pixel")
+
+
+def test_grfilter_dataarray_in_dataarray_out(grid):
+    """
+    grdfilter an input DataArray, and output as DataArray.
+    """
+    # grid = load_earth_relief(registration="pixel")
```
Comment (seisman) [inline pygmt/tests/test_grdfilter.py:25]:
```suggestion
```

Comment (willschlitzer) [reply pygmt/tests/test_grdfilter.py:25]: Thanks, forgot to remove it after loading the pytest fixture

File: pygmt/tests/test_grdfilter.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    result = grdfilter(grid=grid, filter="g600", distance="4")
+    # check information of the output grid
+    assert isinstance(result, xr.DataArray)
+    assert result.coords["lat"].data.min() == -89.5
+    assert result.coords["lat"].data.max() == 89.5
+    assert result.coords["lon"].data.min() == -179.5
+    assert result.coords["lon"].data.max() == 179.5
+    assert result.sizes["lat"] == 180
+    assert result.sizes["lon"] == 360
```
Comment (seisman) [inline pygmt/tests/test_grdfilter.py:33]: Could you also check the minimum and maximum values of the output grid? The current checks can not guarantee that the output grid is correctly filtered.

**Comment by seisman** [comment]
It would be better if you can add two more checks:

- grid file in, grid file out
- grid file in, xr.DataArray out.

File: pygmt/tests/test_grdfilter.py (hunk 2 above)
Comment (willschlitzer) [reply pygmt/tests/test_grdfilter.py:33]: I kept getting errors when I tested those. I'm not sure if there were more decimal places than were being displayed when I printed them, which may have caused the mismatch. I'll give it another try.

Comment (seisman) [reply pygmt/tests/test_grdfilter.py:33]: You can use some numpy testing functions (https://numpy.org/doc/stable/reference/routines.testing.html).

Comment (willschlitzer) [reply pygmt/tests/test_grdfilter.py:33]: The function `assert_almost_equal` worked! Thanks.

**Comment by willschlitzer** [comment]
@seisman Added the two additional tests.

File: pygmt/tests/test_grdfilter.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        result = grdfilter(grid, outgrid=tmpfile.name, filter="g600", distance="4")
+        assert result is None  # grdfilter returns None if output to a file
+        result = grdinfo(tmpfile.name, C=True)
+        assert (
+            result == "-180 180 -90 90 -6147.47265625 5164.11572266 1 1 360 180 1 1\n"
+        )
+
+
+def test_grfilter_file_in_dataarray_out(grid):
```
Comment (seisman) [inline pygmt/tests/test_grdfilter.py:53]: `grid` is unused.
```suggestion
def test_grfilter_file_in_dataarray_out():
```

File: pygmt/tests/test_grdfilter.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    """
+    result = grdfilter(grid=grid, filter="g600", distance="4")
+    # check information of the output grid
+    assert isinstance(result, xr.DataArray)
+    assert result.coords["lat"].data.min() == -89.5
+    assert result.coords["lat"].data.max() == 89.5
+    assert result.coords["lon"].data.min() == -179.5
+    assert result.coords["lon"].data.max() == 179.5
+    np.testing.assert_almost_equal(result.data.min(), -6147.47265625, decimal=2)
+    np.testing.assert_almost_equal(result.data.max(), 5164.1157, decimal=2)
```
Comment (seisman) [inline pygmt/tests/test_grdfilter.py:35]: We usually import in this way `import numpy.testing as npt`, and call functions like
```
npt.assert_almost_equal(result.data.min(), -6147.47265625, decimal=2)
```

See `pygmt/tests/test_info.py` for examples.

File: pygmt/tests/test_grdfilter.py (hunk 3 above)
Comment (willschlitzer) [reply pygmt/tests/test_grdfilter.py]: Bad copy and paste on my part

File: pygmt/tests/test_grdfilter.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
```
Comment (seisman) [inline pygmt/tests/test_grdfilter.py:36]:
```suggestion
    npt.assert_almost_equal(result.data.min(), -6147.47265625, decimal=2)
    npt.assert_almost_equal(result.data.max(), 5164.1157, decimal=2)
```

Comment (willschlitzer) [reply pygmt/tests/test_grdfilter.py:36]: Hoping I don't cause an issue, but I'll commit this, but it looks like it was already changed to `npt` in 1b0dc75

Comment (seisman) [reply pygmt/tests/test_grdfilter.py]: `np.testing.assert_almost_equal` is used in two tests, and 1b0dc75ccdae933cc32e5933b673621957570f85 only changed the 2nd one.

Comment (willschlitzer) [reply pygmt/tests/test_grdfilter.py]: Whoops! Just forgetfulness on my part.

**Review by seisman — APPROVED** [review]
This PR looks good to me. 

I'll keep this PR unmerged for two more days so that @GenericMappingTools/python @GenericMappingTools/python-maintainers can give it a final review.

**Review by weiji14 — APPROVED** [review]
Looks great!

**Pull request merged**